EIPS/eip-7951.md

```diff
@@ -0,0 +1,278 @@
+
+2. **Modular comparison**: The original comparison `r' == r` should be `r' ≡ r (mod n)` to handle cases where the x-coordinate of R' exceeds the curve order n. This ensures mathematically correct verification according to ECDSA standards.
+
+### Verification vs Recovery
+
+This specification uses signature verification rather than public key recovery (like `ECRECOVER`) because:
+
+- Most secp256r1 implementations in hardware and software verify signatures directly
+- The NIST FIPS 186-5 standard[^2] specifies verification, not recovery
+- Verification is more efficient than recovery for this curve
+- Existing hardware implementations provide verification interfaces
+
+### Gas Cost Justification
+
+The 3450 gas cost maintains consistency with RIP-7212 implementations and reflects the computational cost relative to existing precompiles. Benchmarking shows secp256r1 verification is approximately 15% more expensive than secp256k1 operations, justifying the cost relative to `ECRECOVER`.
+
+## Backwards Compatibility
+
+This EIP maintains full interface compatibility with RIP-7212 implementations deployed on Layer 2 networks. The same contract bytecode that works with RIP-7212 will work with this specification.
+
+The security fixes are transparent to correctly implemented callers - they only affect edge cases that should have failed verification anyway. No existing valid use cases are broken.
+
+### Interface Compatibility
+
+The precompile maintains the exact same interface as RIP-7212 to ensure compatibility with existing Layer 2 deployments:
+
+- Same address: `0x100`
+- Same input format: 160 bytes
+- Same output format: 32 bytes
+- Same gas cost: 3450 gas
+- Same return values
+
+## Test Cases
+
+A set of test vectors for verifying implementations is located in a separate [file](../assets/eip-xxxx/test-vectors.json).
+
+## Reference Implementation
+
+A reference implementation is not supplied due to the ubiquity of secp256r1, if however a reference is needed, the NIST specification in FIPS 186-5[^2]
+
+## Security Considerations
+
+### Cryptographic Security
+
+The secp256r1 curve provides approximately 128 bits of security, equivalent to secp256k1 already in use in Ethereum. The curve parameters are standardized by NIST SP 800-186[^1] and have undergone extensive cryptographic analysis by the wider cryptographic community in addition to this curve being deployed to several L2s in the form of RIP-7212.
+
+### Malleability
+
+Unlike secp256k1 ECDSA signatures, secp256r1 ECDSA signatures are not required to be non-malleable per NIST FIPS 186-5 standard [^2]. Applications requiring non-malleability should implement additional checks at the application layer.
+
+### Side-Channel Resistance
+
+We explicitly state that this precompile **IS NOT REQUIRED** to perform all the operations using constant time algorithms.
+
+## Copyright
+
+Copyright and related rights waived via [CC0](../LICENSE.md).
+
+[^1]:
+```csl-json
```

Comment:
some lint errors here still

Reply:
The markdown linter is a bit too strict with these blocks, unfortunately. EIP-1 shows how to mute the warning:

https://github.com/ethereum/EIPs/blob/92639db6cbeda20eabb20212c291410cf0d60d5f/EIPS/eip-1.md?plain=1#L472-L504

Reply:
Thanks @SamWilsn 👍, I've muted the warnings as requested. :)